### PR TITLE
arch: riscv: add an option for empty spurious interrupt handler

### PR DESCRIPTION
**Pull request by magp-nordic**
Add the possibility to disable fault handling in spurious interrupt handler on RISCs and replacce it with an infinite loop.

### CODE REVIEW

**Review by ycsin — COMMENTED** [review]
Q: What improvement do you get from this?

**Comment by magp-nordic** [comment]
@ycsin It reduces code size in some cases, for example, when multithreading and timers (`CONFIG_SYS_CLOCK_EXISTS`) are disabled.

**Comment by fkokosinski** [comment]
Just a thought - if we're adding such functionality here, maybe we should expand it to cover all architectures?

**Comment by magp-nordic** [comment]
> Just a thought - if we're adding such functionality here, maybe we should expand it to cover all architectures? This way we'd have a common kernel configuration mechanism for this.

@fkokosinski Sure, I think this is a good idea, but maybe it could be done in a separate PR? Alternatively, I could modify this PR to make the Kconfig option global and leave implementation only for RISC-V. What do you think?

**Comment by fkokosinski** [comment]
> > Just a thought - if we're adding such functionality here, maybe we should expand it to cover all architectures? This way we'd have a common kernel configuration mechanism for this.
> 
> @fkokosinski Sure, I think this is a good idea, but maybe it could be done in a separate PR? Alternatively, I could modify this PR to make the Kconfig option global and leave implementation only for RISC-V.

From my standpoint the latter sounds better to me. This would probably have to have a pair of Kconfig options like `CONFIG_ARCH_SUPPORTS_EMPTY_IRQ_SPURIOUS` selected by `CONFIG_RISCV`, and `CONFIG_EMPTY_IRQ_SPURIOUS` which would be exposed to the user. But this of course will need a broader discussion here :)

**Comment by magp-nordic** [comment]
> From my standpoint the latter sounds better to me. This would probably have to have a pair of Kconfig options like `CONFIG_ARCH_SUPPORTS_EMPTY_IRQ_SPURIOUS` selected by `CONFIG_RISCV`, and `CONFIG_EMPTY_IRQ_SPURIOUS` which would be exposed to the user. 

Sounds good to me.

> But this of course will need a broader discussion here :)

@fkokosinski What do you mean by broader discussion? Should I involve someone else in the discussion? If so, who should I ping?

Sorry for responding after such a long time, I was on vacation ☺️

**Comment by fkokosinski** [comment]
> What do you mean by broader discussion? Should I involve someone else in the discussion? If so, who should I ping?

I just meant a broader discussion will happen here in the PR when you modify some general kernel Kconfig options :) People will be pinged automatically when you push such changes, so you don't have to ping anyone manually

File: arch/Kconfig [hunk 1]
```diff
@@ -597,6 +598,15 @@ config SIMPLIFIED_EXCEPTION_CODES
 	  down to the generic K_ERR_CPU_EXCEPTION, which makes testing code
 	  much more portable.
 
+
```
Comment (fkokosinski) [inline arch/Kconfig:601]: Nit: an unnecessary blank line added

File: arch/riscv/core/irq_manage.c [hunk 2]
```diff
@@ -37,6 +43,7 @@ FUNC_NORETURN void z_irq_spurious(const void *unused)
 	}
 #endif
 	z_riscv_fatal_error(K_ERR_SPURIOUS_IRQ, NULL);
+#endif
```
Comment (fkokosinski) [inline arch/riscv/core/irq_manage.c:46]:
```suggestion
#endif /* CONFIG_EMPTY_IRQ_SPURIOUS */
```

**Review by fkokosinski — COMMENTED** [review]
Small comments; LGTM otherwise

**Review by fkokosinski — APPROVED** [review]
RISC-V changes LGTM

**Review by masz-nordic — APPROVED** [review]

**Review by jaz1-nordic — APPROVED** [review]

**Pull request merged**